atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/keyvalue/impl/RangeRequestsTest.java

```diff
@@ -104,6 +105,45 @@ public void rangeWithSameBoundIsEmpty() {
         assertThat(RangeRequests.isExactlyEmptyRange(BYTES_2, BYTES_2)).isTrue();
     }
 
+    @Test
+    public void endNameForPrefixScanOfEmptyByteArrayIsEmpty() {
+        assertThat(RangeRequests.createEndNameForPrefixScan(new byte[] {})).isEmpty();
+    }
+
+    @Test
+    public void endNameForPrefixScanOfSingleByteArraysIncrementsTheByte() {
+        assertThat(RangeRequests.createEndNameForPrefixScan(new byte[] {0})).isEqualTo(new byte[] {1});
+        assertThat(RangeRequests.createEndNameForPrefixScan(new byte[] {42})).isEqualTo(new byte[] {43});
+        assertThat(RangeRequests.createEndNameForPrefixScan(new byte[] {127})).isEqualTo(new byte[] {-128});
```

Comment:
Let's leave a small comment explaining why this is the case :)

Reply:
yep, makes sense